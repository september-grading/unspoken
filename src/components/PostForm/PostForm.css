.post-form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: auto;
}

textarea {
  margin: 10px;
  width: 500px;
  height: 100px;
  resize: vertical;
}

input {
  width: 500px;
  margin: 10px;
}

label {
  margin-left: 10px;
}
