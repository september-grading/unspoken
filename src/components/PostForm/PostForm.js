import React from 'react';
import './PostForm.css';

export default function PostForm({ handleSubmit, updatePost, post }) {
  return (
    <>
      <form className="post-form">
        <h1>create post</h1>
        <input
          placeholder="post title..."
          value={post.post_title}
          name="title"
          type="text"
          onInput={(e) => updatePost('post_title', e.target.value)}
        />
        <textarea
          placeholder="tell us more... (minimum 6 characters)"
          value={post.post_content}
          name="content"
          type="text"
          onInput={(e) => updatePost('post_content', e.target.value)}
        />
        <button onClick={handleSubmit} className="edit-button">
          Submit
        </button>
      </form>
    </>
  );
}
