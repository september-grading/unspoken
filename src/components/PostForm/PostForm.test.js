import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import PostForm from './PostForm';

test('renders blank post form', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/create']}>
      <Route path="/create">
        <PostForm post={''} />
      </Route>
    </MemoryRouter>
  );
  await screen.findByText('create post');
  expect(container).toMatchSnapshot();
});
