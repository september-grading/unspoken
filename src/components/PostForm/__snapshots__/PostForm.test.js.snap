// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders blank post form 1`] = `
<div>
  <form
    class="post-form"
  >
    <h1>
      create post
    </h1>
    <input
      name="title"
      placeholder="post title..."
      type="text"
      value=""
    />
    <textarea
      name="content"
      placeholder="tell us more... (minimum 6 characters)"
      type="text"
    />
    <input
      type="submit"
    />
  </form>
</div>
`;
